builders/macos-python-builder.psm1

```diff
@@ -35,6 +35,13 @@ class macOSPythonBuilder : NixPythonBuilder {
         $configureString += " --enable-shared"
         $configureString += " --with-lto"
 
+        ### For Python versions which support it, compile a universal2 (arm64 + x86_64 hybrid) build. The arm64 slice
+        ### will never be used itself by a Github Actions runner but using a universal2 Python is the only way to build
+        ### universal2 C extensions and wheels.
+        if ($this.Version -ge "3.9.1") {
```

Comment:
May I ask why the Python version need to be greater than `3.9.1`? In official Python download page, you can have universal Python for version `3.8.10`. See below:

https://www.python.org/downloads/macos/

Reply:
Yeah, you're right. I hadn't realised that they backported to Python 3.8.x as well. Note though that Python `3.9.0` does not support `universal2` so the conditional would have to be:
```powershell
if ($this.Version -ge "3.8.10" && $this.Version -ne "3.9.0") {
```
(assuming that that is the correct powershell syntax).

Reply:
I think `3.8.10` is the only `3.8` version with `universal2`. So a `==3.8.10` and `>=3.9.1` will work I guess.

Reply:
It can compile `universal2` for `3.8.11` - it's just by that point, python.org stops providing precompiled installers of any kind.